libs/server/Custom/CustomRespCommands.cs

```diff
@@ -194,5 +195,152 @@ private bool TryCustomObjectCommand<TGarnetApi>(GarnetObjectType objType, byte s
 
             return true;
         }
+
+        /// <summary>Parse custom raw string command</summary>
+        /// <param name="cmd">Command name</param>
+        /// <param name="customCommand">Parsed raw string command</param>
+        /// <returns>True if command found, false otherwise</returns>
+        public bool ParseCustomRawStringCommand(string cmd, out CustomRawStringCommand customCommand) =>
+            storeWrapper.customCommandManager.Match(new ReadOnlySpan<byte>(Encoding.UTF8.GetBytes(cmd)), out customCommand);
+
+        /// <summary>Parse custom object command</summary>
+        /// <param name="cmd">Command name</param>
+        /// <param name="customObjCommand">Parsed object command</param>
+        /// <returns>True if command found, false othrewise</returns>
+        public bool ParseCustomObjectCommand(string cmd, out CustomObjectCommand customObjCommand) =>
+            storeWrapper.customCommandManager.Match(new ReadOnlySpan<byte>(Encoding.UTF8.GetBytes(cmd)), out customObjCommand);
+
+        /// <summary>Execute a specific custom raw string command</summary>
+        /// <typeparam name="TGarnetApi"></typeparam>
+        /// <param name="storageApi"></param>
+        /// <param name="customCommand">Custom raw string command to execute</param>
+        /// <param name="key">Key param</param>
+        /// <param name="args">Args to the command</param>
+        /// <param name="output">Output from the command</param>
+        /// <returns>True if successful</returns>
+        public bool InvokeCustomRawStringCommand<TGarnetApi>(ref TGarnetApi storageApi, CustomRawStringCommand customCommand, ArgSlice key, ArgSlice[] args, out ArgSlice output)
+            where TGarnetApi : IGarnetAdvancedApi
+        {
+            ArgumentNullException.ThrowIfNull(customCommand);
+
+            var sbKey = key.SpanByte;
+
+            var inputArg = customCommand.expirationTicks > 0 ? DateTimeOffset.UtcNow.Ticks + customCommand.expirationTicks : customCommand.expirationTicks;
+            var sessionParseState = new SessionParseState();
```

Comment:
No need to instantiate a new parse state, you can use the session's parseState field.

Reply:
The CustomProcedure Execute method's CustomProcedureInput parameter still holds a reference to the session's parseState field and could be used by the CustomProcedure's Execute implementation whenever methods like GetNextArg are called which could happen after the current execution of InvokeCustomRawStringCommand needing to retain the session's parseState field data.

Reply:
Perhaps we can then have a 2nd parse state field in the session to be lazily instantiated and re-used instead of creating a new instance at each invocation?